colour/algebra/interpolation.py

```diff
@@ -31,23 +37,598 @@
 __status__ = 'Production'
 
 __all__ = [
+    'kernel_nearest_neighbour', 'kernel_linear', 'kernel_sinc',
+    'kernel_lanczos', 'kernel_cardinal_spline', 'KernelInterpolator',
     'LinearInterpolator', 'SpragueInterpolator', 'CubicSplineInterpolator',
-    'PchipInterpolator', 'lagrange_coefficients'
+    'PchipInterpolator', 'NullInterpolator', 'lagrange_coefficients'
 ]
 
 
+def kernel_nearest_neighbour(x):
+    """
+    Returns the *nearest-neighbour* kernel evaluated at given samples.
+
```

Comment:
Should we add a more verbose explanation of what this means?

Reply:
Maybe it might also suffice to have a top-level docstring that explains what the kernels represent and how they are used.

Reply:
Shall we put that documentation in the `colour.KernelInterpolator` class?

Reply:
Maybe based on that: 
![image](https://user-images.githubusercontent.com/99779/32824115-92e29412-ca44-11e7-83a7-ddccb0ecb73e.png)

Reply:
E.g.:

```python
class KernelInterpolator(object):
    """
    Kernel based interpolation of a 1-D function.

    The reconstruction of a continuous signal can be described as a linear
    convolution operation. Interpolation can be expressed as a convolution of
    the given discrete function :math:`g(x)` with some continuous interpolation
    kernel :math:`k(w)`: [1]_ [2]_

    :math:`\hat{g}(w_0) = [k * g](w_0) = \sum_{x=-\infty}^{\infty}k(w_0 - x)\cdot g(x)`

    [...]
    """
```

Reply:
Ahh, now I get it. It's *that* kind of kernel. That took longer than it probably should have. Yeah, in that case, I think pointing to the mathematical nature of the kernel functions and their requirements (being functions of the type float->float with Domain -inf-->+inf but needs to support arrays, will be used in convolution operation) should be quite helpful.  I like your doc sample.

Reply:
Technically the method could be extended to n-dimensional arrays, e.g. images, but we don't really need it now.